### DISCUSSION TITLE
Using AWC client inside tauri application

### DISCUSSION THREAD

**Discussion by RProduction** [actix/actix-web#3129] [Q&A] [answered]
Hello I have some question regarding AWC crate client library. Can I use AWC inside tauri? Do I need to run full actix-web server for that? Is there is a way to spawn actix-rt thread in another separate thread? Or AWC is indeed thightly coupled to actix-web?

Edit:
I saw the crate dependencies inside AWC crate. Seem like actix-rt is already included, is that mean it will just work? I just need to use macro and it is done?

**Comment by robjtede** [comment, marked as answer]
Firstly, you don't need a server to run `awc`.

You can use the actix-rt macro to get a usable secondary thread.

```rust
// this macro expands to a sync function
#[actix_rt::main]
async fn non_main_thread() {
   let client = awc::Client::default();
}

std::thread::spawn(move || non_main_thread());
```

**Comment by RProduction** [comment]
Thank you for fast answer. Seem like it possible to do that. I am going to try it and come back here later.
